Reword license mentions (#572)
mattermost/mattermost-plugin-incident-collaboration: README.md

[![Code Coverage](https://img.shields.io/codecov/c/github/mattermost/mattermost-plugin-incident-collaboration/master.svg)](https://codecov.io/gh/mattermost/mattermost-plugin-incident-collaboration)
[![Release](https://img.shields.io/github/v/release/mattermost/mattermost-plugin-incident-collaboration)](https://github.com/mattermost/mattermost-plugin-incident-collaboration/releases/latest)

Mattermost Incident Collaboration allows your team to coordinate, manage, and resolve incidents from within Mattermost. For configuration and administration information visit our [documentation](https://docs.mattermost.com/administration/devops-command-center.html).

![Mattermost Incident Collaboration](assets/incident_response.png)

> ## License

Reviewer: @jamiehurewitz when you're reviewing the source available license text, are you also open to reviewing this License section of the README file to ensure we're not saying anything incorrectly?

Reviewer 2: Yes.

Reviewer 2: @jasonblais @agarciamontoro I think it is ok to go ahead and make the replacement terms, I can just incorporate that as part of my review.

Author: @jamiehurewitz, cool, thank you! Then I'll merge this. Please ping us if you need anything from us :)

This repository is licensed under the [Mattermost Source Available License](LICENSE). See [frequently asked questions](https://docs.mattermost.com/overview/faq.html#mattermost-source-available-license) to learn more.

> Although a valid Mattermost Enterprise Edition E20 license is required to access all features if using this plugin in production, the [Mattermost Source Available License](LICENSE) allows you to compile and test this plugin in development and testing environments without a Mattermost Enterprise Edition E20 license. As such, we welcome community contributions to this plugin.

Reviewer 3: ```suggestion
Although a valid Mattermost Enterprise license is required to access all features if using this plugin in production, the [Mattermost Source Available License](LICENSE) allows you to compile and test this plugin in development and testing environments without a Mattermost Enterprise license. As such, we welcome community contributions to this plugin.
```

Reviewer 3: Do community members need any type of license to contribute or can they be on the starter package?

Author: They can be on the Starter package, or even in the Team Edition. If they enable the development and testing modes in the system console, they get access to all features. I like your suggestion.

> If you're running an Enterprise Edition of Mattermost and don't already have a valid license, you can obtain a trial license from **System Console > Edition and License**. If you're running the Team Edition of Mattermost, including when you run the server directly from source, you may instead configure your server to enable both testing (`ServiceSettings.EnableTesting`) and developer mode (`ServiceSettings.EnableDeveloper`). These settings are not recommended in production environments. See [Contributing](#contributing) to learn more about how to set up your development environment.

Reviewer 3: ```suggestion
If you're running Mattermost Professional or Mattermost Enterprise and don't already have a valid license, you can obtain a trial license from **System Console > Edition and License**. If you're running Mattermost Starter, including when you run the server directly from source, you may instead configure your server to enable both testing (`ServiceSettings.EnableTesting`) and developer mode (`ServiceSettings.EnableDeveloper`). These settings are not recommended in production environments. See [Contributing](#contributing) to learn more about how to set up your development environment.
```

Author: If you're running an Enterprise Edition of Mattermost and don't already have a valid license <- This means the Starter Edition, not the professional one. We can change it to that.

I think we still have the Team Edition (don't know if its name changed, though), which is functionally identical to Starter, but technically different (they're different binaries: in Team Edition, you cannot add a license; in Starter, you can upgrade to Professional or Enterprise by just updating the license).

Reviewer 3: Ah I see what you mean. Thanks for clarifying that!

Author: @justinegeffen I am re-reading this and I don't know what I meant yesterday. Your suggestion is perfectly valid. Sorry for that!
